### PR TITLE
add ScaleIO storage software data collection

### PR DESCRIPTION
**Pull request by luojiagen**
**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: delfin/drivers/dell_emc/scaleio/rest_handler.py [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+                            host_json.get('id'),
+                        "os_type": host_json.get('osType'),
+                        "status": status,
+                        "ip_address": ip_address
+                    }
+                    host_list.append(host_dict)
+                return host_list
+        except Exception as e:
+            LOG.error("Get Storage hosts error: %s", six.text_type(e))
+            raise e
```
Comment (ghxiaobo) [inline delfin/drivers/dell_emc/scaleio/rest_handler.py:313]: Exceptions cannot be thrown directly

File: delfin/drivers/dell_emc/scaleio/rest_handler.py [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+            else:
+                LOG.error("Login error. URL: %(url)s\n"
+                          "Reason: %(reason)s.",
+                          {"url": consts.REST_AUTH_LOGIN, "reason": res.text})
+                if 'User authentication failed' in res.text:
+                    raise exception.InvalidUsernameOrPassword()
+                else:
+                    raise exception.StorageBackendException(
+                        six.text_type(res.text))
+        except Exception as e:
```
Comment (ghxiaobo) [inline delfin/drivers/dell_emc/scaleio/rest_handler.py:55]: Exceptions cannot be thrown directly

File: delfin/drivers/dell_emc/scaleio/rest_handler.py [hunk 3]
```diff
@@ -0,0 +1,354 @@
...
+    def list_alerts(self, query_para=None):
+        alert_list = []
+        try:
+            storage_alert = self.get_rest_info(consts.REST_SCALEIO_ALERT)
+            if storage_alert:
+                for json_alert in storage_alert:
+                    match_key = json_alert.get('id') + json_alert.get('name')
+                    occur_time = json_alert.get('startTime')
+                    datetime_obj = datetime.datetime.strptime(
+                        occur_time, "%Y-%m-%dT%H:%M:%S.%fZ")
```
[the remark this answers is not in the document]
Comment (luojiagen) [reply delfin/drivers/dell_emc/scaleio/rest_handler.py:230]: The time format parameter has been changed to a constant
image

File: delfin/drivers/dell_emc/scaleio/rest_handler.py [hunk 4]
```diff
@@ -0,0 +1,354 @@
...
+        except Exception as e:
+            err_msg = "Logout error: %s" % (six.text_type(e))
+            LOG.error(err_msg)
+            raise e
+
+    def get_storage(self):
+        try:
+            storage_json = self.get_rest_info(consts.REST_SCALEIO_SYSTEM)
+            if storage_json:
+                for system_json in storage_json:
```
[the remark this answers is not in the document]
Comment (luojiagen) [reply delfin/drivers/dell_emc/scaleio/rest_handler.py:72]: Judgment has been changed
$MD}D7X57B JV7N@SB3(_52

File: delfin/drivers/dell_emc/scaleio/rest_handler.py [hunk 5]
```diff
@@ -0,0 +1,354 @@
...
+            LOG.error("Get Storage System error: %s", six.text_type(e))
+            raise e
+
+    def list_storage_pools(self, storage_id):
+        storage_pool_list = []
+        try:
+            storage_pool_json = self.get_rest_info(
+                consts.REST_SCALEIO_STORAGE_POOL)
+            if storage_pool_json:
+                for pool_json in storage_pool_json:
```
[the remark this answers is not in the document]
Comment (luojiagen) [reply delfin/drivers/dell_emc/scaleio/rest_handler.py:127]: Judgment has been changed
image

File: delfin/drivers/dell_emc/scaleio/rest_handler.py [hunk 6]
```diff
@@ -0,0 +1,354 @@
...
+            LOG.error("Get Storage pool error: %s", six.text_type(e))
+            raise e
+
+    def list_volumes(self, storage_id):
+        list_volumes = []
+        try:
+            storage_volume_json = self.get_rest_info(
+                consts.REST_SCALEIO_VOLUMES)
+            if storage_volume_json:
+                for json_volume in storage_volume_json:
```
[the remark this answers is not in the document]
Comment (luojiagen) [reply delfin/drivers/dell_emc/scaleio/rest_handler.py:165]: Judgment has been changed
image

File: delfin/drivers/dell_emc/scaleio/rest_handler.py [hunk 7]
```diff
@@ -0,0 +1,354 @@
...
+        except Exception as e:
+            LOG.error("Get Storage volume error: %s", six.text_type(e))
+            raise e
+
+    def list_disks(self, storage_id):
+        disks_list = []
+        try:
+            disks_json = self.get_rest_info(consts.REST_SCALEIO_DISKS)
+            if disks_json:
+                for json_disk in disks_json:
```
[the remark this answers is not in the document]
Comment (luojiagen) [reply delfin/drivers/dell_emc/scaleio/rest_handler.py:199]: Judgment has been changed
image

File: delfin/drivers/dell_emc/scaleio/rest_handler.py [hunk 8]
```diff
@@ -0,0 +1,354 @@
...
+        except Exception as e:
+            LOG.error("Get Storage disk error: %s", six.text_type(e))
+            raise e
+
+    def list_alerts(self, query_para=None):
+        alert_list = []
+        try:
+            storage_alert = self.get_rest_info(consts.REST_SCALEIO_ALERT)
+            if storage_alert:
+                for json_alert in storage_alert:
```
[the remark this answers is not in the document]
Comment (luojiagen) [reply delfin/drivers/dell_emc/scaleio/rest_handler.py:226]: Judgment has been changed
image

File: delfin/drivers/dell_emc/scaleio/rest_handler.py (hunk 2 above)
Comment (luojiagen) [reply delfin/drivers/dell_emc/scaleio/rest_handler.py:55]: Throw the corresponding exception
image

File: delfin/drivers/dell_emc/scaleio/rest_handler.py (hunk 1 above)
Comment (luojiagen) [reply delfin/drivers/dell_emc/scaleio/rest_handler.py]: Throw the corresponding exception
image

**Pull request merged**